### ISSUE TITLE
Multiple interdependent axes

### ISSUE THREAD

**Issue by kricka91** [plotly/plotly.js#5076]
I would like to be able to create two y axes in a diagram, which are used for the same series, describing the y value in for example different units, such as in this example:
https://i.stack.imgur.com/ROFCo.png

As far as I can tell, there is currently no way of doing this the proper way (please correct me if I'm wrong). I don't think creating a duplicate series is an acceptable workaround as I don't want it to show up in the legend and I don't want the user to be able to hide the series and thus mess up one of the axes.

I would prefer there to be an option to link an axis to another axis rather than to a data series. The axes should remain in sync at all times as the user interacts in whatever way with the diagram.

This could theoretically be extended two more than two y axes, or to multiple x axes.

(Also, if you have any suggestions for good workarounds in the meantime, I would appreciate it.)

**Comment by nicolaskruchten** [comment]
We have support for this kind of thing via "fixed ratio axes" and "matched axes": https://plotly.com/javascript/axes/#fixedratio-axes and https://plotly.com/javascript/reference/layout/xaxis/#layout-xaxis-matches ... Do either of those meet your needs?

**Comment by kricka91** [comment]
I've been trying them out for a bit and I eventually found a workaround, albeit not a very pretty one.

With **fixedratio-axes** I am able to keep a fixed ratio between a certain distance unit on the y axes, but the user can still scroll one of the axes without affecting the other, so that they become misaligned.

The **matches** option is interesting and does keep the axes in sync. However, all the values will match exactly, whereas I am interested in one axis being an "upscaled" version of the other. This feels, intuitively, like something that shouldn't be too hard to implement by extending the "matches" property. **Matches** seems to override the **scaleanchor** and **scaleratio** properties, so I cannot use them together.

It is possible, though, to use the **matches** property and formatting (https://github.com/d3/d3-format/blob/master/README.md) the text differently on different axes. By overriding the d3 locale formatter (https://github.com/plotly/plotly.js/issues/1464#issuecomment-498050894), I was able to simply format a number as a totally different number.


```
<head>
	<!-- Load plotly.js into the DOM -->
	<script src='https://cdn.plot.ly/plotly-latest.min.js'></script>
</head>
<body>
	<div id='myDiv'><!-- Plotly chart will be drawn inside this DIV --></div>
</body>
```

```
var org_locale = Plotly.d3.locale;
Plotly.d3.locale = (locale) => {
  var result = org_locale(locale);
  var org_number_format = result.numberFormat;
  result.numberFormat = (format) => {
    const splits = format.split("_");
    if (splits[0] === "scale") {
      const k = parseFloat(splits[1]);
      const oldFormat = (splits.length > 2) ? splits[2] : "";
      return (x) => {
        return org_number_format(oldFormat)(x*k);
      }
    }
    else {return org_number_format(splits[0])}
  }
  return result;
}

var trace0 = {
  x: [0,1,2,3,4],
  y: [1,2,4,8,16],
  yaxis:"y"
}

var data = [trace0,{yaxis: "y2"}]

var layout = {
    width: 800,
    title: "fixed-ratio axes",
    yaxis: {
      title: "meters",
      rangemode: "tozero",
    },
    yaxis2: {
      matches: "y",
      title: "feet",
      side: "right",
      overlaying: "y",
      tickformat: "scale_"+(1/0.3048)+"_.2f"
    },
}

Plotly.newPlot('myDiv', data, layout)

```

Thanks a lot for the help, but I still think there is room for improvement so you don't have to rely on deceitful formatting.

**Comment by nicolaskruchten** [comment]
So this is pretty close, and works when box-zooming and panning: https://codepen.io/nicolaskruchten/pen/xxVOEaQ?editors=0010

I notice that grabbing and dragging the y-axis doesn't cause the y2-axis to pan, however... @alexcjohnson @archmoj is this a bug or is the box-zooming/panning a happy accident in this case?

**Comment by alexcjohnson** [comment]
That system is working as intended - when you zoom or pan via the plot area, there's nothing to tell you which specific axes on that area you wanted to alter, so it alters them all. But when you zoom or pan via one of the individual axes, we only change that one. This can be quite useful if, for example, you have several traces measuring different quantities entirely but you want to see if their features match up, you can slide and scale one of them over top of the other.

@kricka91 you make a good argument for why one might want `matches` and `scaleratio` together - though the word "matches" is a bit... stretched... if we continue to use that. But there's more to it in the general case, you may want an offset as well, like Celsius / Fahrenheit.

**Comment by nicolaskruchten** [comment]
Seems like "matches" is the wrong thing to use here... we probably just need a way to pin one axis' 0 to a level on the other. That would help with or without scaleratio in a few different circumstances actually.

**Comment by alexcjohnson** [comment]
> pin one axis' 0 to a level on the other ... without scaleratio

While still allowing independent scaling (via zoom interactions)? This would be interesting, but my gut reaction is much more complicated than a fixed mapping (just a more generic linear mapping than the identity of `matches`), as it would imply various weird behavior when panning or zooming one of the axes.

**Comment by nicolaskruchten** [comment]
Yeah... someone recently brought up dual y-axes with matching 0=0 but otherwise independent scales, on the forum I think.

**Comment by jackparmer** [comment]
This issue has been tagged with `NEEDS SPON$OR`

A [community PR](https://github.com/plotly/plotly.js/pulls?q=is%3Apr+is%3Aopen+label%3A%22type%3A+community%22) for this feature would certainly be welcome, but our experience is deeper features like this are difficult to complete without the Plotly maintainers leading the effort.

**Sponsorship range: $15k-$20k**

What Sponsorship includes:
- Completion of this feature to the Sponsor's satisfaction, in a manner coherent with the rest of the Plotly.js library and API
- Tests for this feature
- Long-term support (continued support of this feature in the latest version of Plotly.js)
- Documentation at [plotly.com/javascript](https://plotly.com/javascript/)
- Possibility of integrating this feature with Plotly Graphing Libraries (Python, R, F#, Julia, MATLAB, etc)
- Possibility of integrating this feature with Dash
- Feature announcement on [community.plotly.com](https://community.plotly.com/tag/announcements) with shout out to Sponsor (or can remain anonymous)
- Gratification of advancing the world's most downloaded, interactive scientific graphing libraries (>50M downloads across supported languages)

Please include the link to this issue when [contacting us](https://plotly.com/get-pricing/) to discuss.

**Comment by thegreencouch** [comment]
@kricka91  Overriding the d3 locale formatter in this manner is not possible anymore with the same code with the new updates(d3.locale does not exist on plotly.js anymore). Do you know if there's an alternate way to do the same patch?

**Comment by gvwilson** [comment]
Hi - we are trying to tidy up the stale issues and PRs in Plotly's public repositories so that we can focus on things that are still important to our community. Since this one has been sitting for a while, I'm going to close it; if it is still a concern, please add a comment letting us know what recent version of our software you've checked it with so that I can reopen it and add it to our backlog. Alternatively, if it's a request for tech support, please post in our community forum. Thank you - @gvwilson

**Comment by JacobHast** [comment]
@gvwilson I would like to request a re-opening of this issue. I'm an experimental researcher and frequently find that I want to visualize data in multiple units (e.g. temperature in C and F, power in W and dBm, etc.), so to me this is quite an essential feature for scientific data-plotting. 
It can be done straightforwardly in matplotlib, https://matplotlib.org/stable/gallery/subplots_axes_and_figures/secondary_axis.html, and I have also been able to achieve it in Bokeh, so plotly appears to be lagging behind here. 

Would greatly appreciate if this can be added to the backlog!
